lib/atlas/initializer_input.rb

```diff
@@ -0,0 +1,23 @@
+module Atlas
+  class InitializerInput
+    include ActiveDocument
+
+    DIRECTORY = 'initializer_inputs'
+
+    attribute :query,           String
+    attribute :share_group,     Symbol
+    attribute :priority,        Integer, default: 0
+    attribute :unit,            String
+    attribute :user_value,      Float
+
+    validates_presence_of :query
+
+    def self.collection_from_hash(init_hash)
+      init_hash.map do |key, user_value|
+        input = find(key)
+        input.user_value = user_value
```

Comment:
I know you've changed this already, but as a general rule don't modify an ActiveDocument at runtime. In a script, Rake task, or console is fine, but not within Atlas code itself or within ETEngine.

Atlas uses an identity map for documents, which means that every time you retrieve a document you get the exact same object back:

```ruby
Atlas::Node.find(:buildings_appliances_electricity).object_id
# => 70249863187440

Atlas::Node.find(:buildings_appliances_electricity).object_id
# => 70249863187440
```

*(in hindsight, I'm less keen on this than I once was; but it is what it is 😆 )*

If you do modify at runtime, it means the application cannot be thread-safe anymore:

```
  | Thread 1                   | Thread 2
  +----------------------------+----------------------------
1 | Set value for input 1 = 10 |
2 | Set value for input 2 = 20 |
3 | Execute input 1 (10)       |
4 |                            | Set value for input 1 = 50
5 |                            | Set value for input 2 = 90
6 | Execute input 2 (90)       |
7 |                            | Execute input 1 (50)
8 | ...                        | ...
```

Step 6 is executed with the wrong value!

In fairness: ETEngine is not currently thread-safe, so this is all academic. But in recent years I made it a personal goal to not add any new code to ETE (or to libraries like Atlas and Refinery) which would take us further from the eventual goal of making it thread-safe.